File: NeetCode/01-Arrays-Hashing/TwoSums.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

class Solution {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int size = scanner.nextInt();

        int[] nums = new int[size];
        for (int i = 0; i < size; i++) {
            nums[i] = scanner.nextInt();
        }

        int target = scanner.nextInt();

        int[] result = twoSum(nums, target);
        
        scanner.close();
    }

    public static int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];

            if (map.containsKey(complement)) {
                return new int[] { map.get(complement), i };
            }
            map.put(nums[i], i);
        }
        return new int[] {}; 
    }
}

File: NeetCode/01-Arrays-Hashing/TwoSums.cpp
#include <iostream>
#include <vector>
#include <unordered_map>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> numsMap;

    for (int i = 0; i < nums.size(); i++) {
        int complement = target - nums[i];

        if (numsMap.find(complement) != numsMap.end()) {
            return {numsMap[complement], i};
        }

        numsMap[nums[i]] = i;
    }
    return {};
    }
};
int main() {
    int size, target;

    cin >> size;
    Solution sol;

    vector<int> nums(size);
    for (int i = 0; i < size; i++) {
        cin >> nums[i];
    }

    cin >> target;

    vector<int> result = sol.twoSum(nums, target);

    return 0;
}